tv/tv.cpp

```diff
@@ -168,6 +194,9 @@ struct TVLegacyPass final : public llvm::ModulePass {
       return false;
     }
 
+    if (!MClone && first && opt_save_ir && !report_filename.empty())
```

Comment:
I think this condition is only true when you run the first optimization and for the first function.
If I run `-O2` and the bug is somewhere in the middle, the original IR (before any optimization) will be written to disk I think.

Reply:
ok so we have to keep saving additional versions each time?

I wrote it like that to avoid cloning the module for every function but I'll find a better way to do that

Reply:
I think it's fine to keep cloning. I think we'll only enable the save-ir flag when we know there's a bug, so it's fine to slow things down a bit.

Reply:
so you're thinking we we-run alive2 with this option after finding a crash? I was hoping to just leave this turned on all the time while bug-hunting...

Reply:
but of course re-running is easy since we have the exact command line. maybe that's a better idea. let me play around with this some more.

Reply:
We need an option that allows to dump the IR right before the buggy optimization + the name of that optimization. That's what we need to reproduce the issue. (we may need some extra pass arguments, but those are probably not easy to recover automatically. Though a human can grab those easily from the overall cmd line I think)

Reply:
well, alternatively we can just log the command line and also the original IR and then narrow things down more later on as part of reducing